pkg/kernel/mcp/ip.go

```diff
@@ -0,0 +1,111 @@
+		return nil, types.Result{}, fmt.Errorf("error while getting ip data: %w", err)
+	}
+	if in.FilterParameters != "" {
+		if err := validateFilterParameters(in.FilterParameters); err != nil {
+			return nil, types.Result{}, fmt.Errorf("error while getting ip data: %w", err)
+		}
+	}
+	if in.Options != "" {
+		if err := validateFilterParameters(in.Options); err != nil {
+			return nil, types.Result{}, fmt.Errorf("error while getting ip data: %w", err)
+		}
+	}
+
+	cmd := newCommand("ip")
+	cmd.addIfNotEmpty(in.Options, in.Options)
+	cmd.add(strings.Fields(in.Command)...)
+	cmd.addIfNotEmpty(in.FilterParameters, strings.Fields(in.FilterParameters)...)
+
+	stdout, err := s.executeCommand(ctx, req, in.Node, in.Image, cmd.build())
+	if err != nil {
+		return nil, types.Result{}, fmt.Errorf("error while getting ip data: %w", err)
+	}
+	stdout = limitOutputLines(stdout, in.MaxLines)
+	return nil, types.Result{Data: stdout}, nil
+}
+
+// ValidateIPCommand validates that the IP command is allowed.
+func validateIPCommand(ipCommand string) error {
+	if _, err := strconv.Atoi(ipCommand); err == nil {
+		return fmt.Errorf("invalid ip command: %s", ipCommand)
+	}
+	splitCommand := strings.Fields(strings.TrimSpace(ipCommand))
+	if len(splitCommand) < 2 {
+		return fmt.Errorf("invalid ip command: %s", ipCommand)
+	}
+	validIpCommand := map[string]bool{
+		"address":     true,
+		"link":        true,
+		"neighbour":   true,
+		"netns":       true,
+		"route":       true,
+		"rule":        true,
+		"vrf":         true,
+		"xfrm state":  true,
+		"xfrm policy": true,
+	}
+
+	// "ip l s" can be used to set a link up or down. This should be considered as an invalid command.
+	if strings.HasPrefix(splitCommand[0], "l") && splitCommand[1] == "s" {
+		return fmt.Errorf("invalid ip command: %s", ipCommand)
+	}
+
+	var valid bool
+	for command := range validIpCommand {
+		commandFields := strings.Fields(command)
+		// Check if the input command matches the expected command pattern
+		// For single-word commands like "address", we check: splitCommand[0] matches "address" and splitCommand[1] matches "show"
+		// For multi-word commands like "xfrm state", we check: splitCommand[0:2] matches ["xfrm", "state"] and splitCommand[2] matches "show"
+
+		if len(commandFields) == 1 {
+			// Single-word command (e.g., "address show", "link show")
+			if strings.HasPrefix(command, splitCommand[0]) && strings.HasPrefix("show", splitCommand[1]) {
+				valid = true
+				break
+			}
+		} else {
+			// Multi-word command (e.g., "xfrm state show", "xfrm policy show")
+			// Need at least len(commandFields) + 1 words (command words + "show")
+			if len(splitCommand) >= len(commandFields)+1 {
+				allMatch := true
+				for i, cmdField := range commandFields {
+					if !strings.HasPrefix(cmdField, splitCommand[i]) {
+						allMatch = false
+						break
+					}
+				}
+				if allMatch && strings.HasPrefix("show", splitCommand[len(commandFields)]) {
```

Comment:
`show` is not a valid command for `ip xfrm`: https://man7.org/linux/man-pages/man8/ip-xfrm.8.html.
`ip xfrm [state|policy] list` is the valid command.